src/reducers/selectorLocations.js

```diff
@@ -0,0 +1,26 @@
+import Immutable from 'immutable';
+
+import {SelectorLocations} from '../records';
+
+const emptyMap = new Immutable.Map();
+
+export default function reduceProjects(stateIn, action) {
+  let state;
+
+  if (stateIn === undefined) {
+    state = emptyMap;
+  } else {
+    state = stateIn;
+  }
+
+  switch (action.type) {
+    case 'UPDATE_SELECTOR_LOCATIONS':
+      return state.set(
+        'selectors',
```

Comment:
I don’t think there’s any need for an additional level of nesting here—isn’t the `SelectorLocations` record itself sufficient to represent this slice of the state?